WikiDrips/WikiTableViewController.swift

```diff
@@ -52,13 +47,19 @@ class WikiTableViewController: UITableViewController, UISearchBarDelegate {
     }
     
     func search() {
-//      TODO: replace with async call to Wikipedia API
-        let someDateTime = Date(timeIntervalSinceNow: -(Double(wikiDocs.count) * 60))
-        let testData: [String: AnyObject] = ["title": "My new title" as AnyObject, "pubDate": someDateTime as AnyObject]
-        if let wikiDoc = WikiDoc(data: testData) {
-            wikiDocs.insert([wikiDoc], at: 0)
+        guard let searchText = searchText else {
+            return
+        }
+        if let wikiRequest = WikiRequest(searchText: searchText) {
+            wikiRequest.fetchWikiDocs { (newDocs) -> Void in
+                DispatchQueue.main.async(){
+                    if newDocs.count > 0 {
+                        self.wikiDocs = wikiRequest.wikiDocs
```

Comment:
Notice that by using `self` here you are strongly capturing the view controller from the request. That's not always a good thing cause request can take a long time to complete.